iceoryx2-bb/derive-macros/tests/placement_default_tests.rs

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) 2024 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Apache Software License 2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0, or the MIT license
+// which is available at https://opensource.org/licenses/MIT.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+#[cfg(test)]
+mod placement_new {
+    use std::{
+        alloc::{alloc, dealloc, Layout},
+        sync::atomic::{AtomicU64, AtomicUsize, Ordering},
+    };
+
+    use iceoryx2_bb_derive_macros::PlacementDefault;
+    use iceoryx2_bb_elementary::placement_default::PlacementDefault;
+    use iceoryx2_bb_testing::assert_that;
+
+    static DEFAULT_CTOR_COUNT: AtomicUsize = AtomicUsize::new(0);
+    static FUU_VALUE: AtomicU64 = AtomicU64::new(0);
+    static BAR_VALUE: AtomicU64 = AtomicU64::new(0);
+
+    #[derive(Copy, Clone)]
+    struct UnitStruct;
+
+    impl PlacementDefault for UnitStruct {
+        unsafe fn placement_default(_ptr: *mut Self) {
+            DEFAULT_CTOR_COUNT.fetch_add(1, Ordering::Relaxed);
+        }
+    }
+
+    struct Fuu(u64);
+
+    impl PlacementDefault for Fuu {
+        unsafe fn placement_default(ptr: *mut Self) {
+            DEFAULT_CTOR_COUNT.fetch_add(1, Ordering::Relaxed);
+            ptr.write(Self(FUU_VALUE.load(Ordering::Relaxed)))
+        }
+    }
+
+    struct Bar {
+        value: u64,
+    }
+
+    impl PlacementDefault for Bar {
+        unsafe fn placement_default(ptr: *mut Self) {
+            DEFAULT_CTOR_COUNT.fetch_add(1, Ordering::Relaxed);
+            ptr.write(Self {
+                value: BAR_VALUE.load(Ordering::Relaxed),
+            })
+        }
+    }
+
+    #[derive(PlacementDefault)]
+    struct NamedTestStruct {
+        value1: UnitStruct,
+        value2: Fuu,
+        value3: Bar,
+    }
+
+    #[derive(PlacementDefault)]
+    struct UnnamedTestStruct(Fuu, Bar, Bar, UnitStruct, UnitStruct, UnitStruct);
+
+    #[derive(PlacementDefault)]
+    struct GenericStruct<T1: PlacementDefault, T2: PlacementDefault> {
+        value1: T1,
+        value2: T2,
+    }
+
+    #[derive(PlacementDefault)]
+    struct GenericUnnamedStruct<T1: PlacementDefault, T2: PlacementDefault>(T1, T2);
+
+    #[test]
+    fn placement_default_derive_for_structs_works() {
+        DEFAULT_CTOR_COUNT.store(0, Ordering::Relaxed);
+        FUU_VALUE.store(123, Ordering::Relaxed);
+        BAR_VALUE.store(456, Ordering::Relaxed);
+
+        let layout = Layout::new::<NamedTestStruct>();
+        let memory = unsafe { alloc(layout) } as *mut NamedTestStruct;
```

Comment:
```suggestion
        let memory = unsafe { alloc_zeroed(layout) } as *mut NamedTestStruct;
```

The same for all other occurrences further down